### PR TITLE
Few improvements towards lab6 (actually the "second round" of the lab5 exercise)

### PR DESCRIPTION
**Pull request by battibass**
To prepare the second round of lab5:

`lab5.md` was updates by:
 - improving few parts of the assignment description;
 - adding a little discussion about the point reached ad the end of the _mandatory part_ of the assignment;
 - adding a couple of optional exercises.

Cosmetics was improved here and there in the case of provided code examples.

### CODE REVIEW

File: labs/lab5/lab5.md [hunk 1]
```diff
@@ -297,8 +303,8 @@ semplificata di quello che si può fare per sviluppare programmi di simulazione
 da utilizzare per studiare sistemi di punti materiali:
 
 1. caratterizzati da un numero "elevato" componenti;
-2. per i quali le iterazioni tra i singoli componenti non sono facilmente
-   risolvibili analiticamente.
+2. per i quali le iterazioni tra i singoli componenti possono non risultare
```
Comment (giacomini) [inline labs/lab5/lab5.md]: iterazioni o interazioni?

File: labs/lab5/lab5.md [hunk 2]
```diff
@@ -477,10 +502,80 @@ combinazione con una funzione `print_state` per la stampa di:
...
+> `std::back_inserter`, quale?
+>
+> __NOTA__: invece di definire `p_x` fuori da `std::generate_n` ed effettuarne
+> il `capture` all'interno della `lambda` che utilizzerete per definire la
+> logica della `generator function` (`g` nella pagina web) potreste
+> utilizzare una `mutable lambda`.
+
+### Utilizzare `std::transform` in `Chain::evolve`
+
+Potete infine provare ad utilizzare utilizzare
```
Comment (giacomini) [inline labs/lab5/lab5.md]:
```suggestion
Potete infine provare ad utilizzare
```

File: labs/lab5/lab5.md [hunk 3]
```diff
@@ -477,10 +502,80 @@ combinazione con una funzione `print_state` per la stampa di:
...
+[`std::transform`](https://en.cppreference.com/w/cpp/algorithm/transform)
+all'interno del metodo `evolve`.
+
+> __NOTA__: leggendo la documentazione, potete notare che esiste una versione di
+> `std::transform` che permette di effettuare un operazioni tra due argomenti
+> diversi dello stesso tipo (`binary_op` nella pagina web). Potete
+> utilizzare questa versione dell'algoritmo per il calcolo della forza relativa
+> all'iterazione di due `PPState`
```
Comment (giacomini) [inline labs/lab5/lab5.md]: interazione

File: labs/lab5/lab5.md [hunk 4]
```diff
@@ -477,10 +502,80 @@ combinazione con una funzione `print_state` per la stampa di:
...
+> utilizzare una `mutable lambda`.
+
+### Utilizzare `std::transform` in `Chain::evolve`
+
+Potete infine provare ad utilizzare utilizzare
+[`std::transform`](https://en.cppreference.com/w/cpp/algorithm/transform)
+all'interno del metodo `evolve`.
+
+> __NOTA__: leggendo la documentazione, potete notare che esiste una versione di
+> `std::transform` che permette di effettuare un operazioni tra due argomenti
```
Comment (giacomini) [inline labs/lab5/lab5.md]: un'operazione o operazioni

File: labs/lab5/lab5.md [hunk 5]
```diff
@@ -477,10 +502,80 @@ combinazione con una funzione `print_state` per la stampa di:
...
+> diversi dello stesso tipo (`binary_op` nella pagina web). Potete
+> utilizzare questa versione dell'algoritmo per il calcolo della forza relativa
+> all'iterazione di due `PPState`
+> successivi.
+>
+> __NOTA__: qualora non lo aveste già fatto nella vostra implementazione di
+> `evolve`, risulta conveniente registrare in una variabile il risultato del
+> calcolo della forza di interazione tra due `PPState` (es. 1 e 2 quando si
+> calcola l'evoluzione del `PPState` 1) per evitare di ricalcolarla quando va
+> applicata, con segno opposto, nell calcolo dell'evoluzione del `PPState`
```
Comment (giacomini) [inline labs/lab5/lab5.md]:
```suggestion
> applicata, con segno opposto, nel calcolo dell'evoluzione del `PPState`
```

File: labs/lab5/lab5.md [hunk 6]
```diff
@@ -477,10 +502,80 @@ combinazione con una funzione `print_state` per la stampa di:
...
+```
+
+> __NOTA__: in questo caso, suggeriamo di assumere che `p_v` sia sempre `0.0`.
+>
+> __NOTA__: come primo argomento  di `std::generate_n` potete utilizzare
+> `std::back_inserter(c_hooke)`, che invoca la funzione `Chain::push_back` ad
+> ogni passo dell'iterazione.
+>
+> __NOTA__: Se non l'avete fatto, sarà necessario aggiungere una piccola
+> dichiarazione `using` per evitare errori di compilazione quando si utilizza
```
Comment (giacomini) [inline labs/lab5/lab5.md]: non ho capito questa cosa di usare `using`

Comment (battibass) [reply labs/lab5/lab5.md]: Ho aggiustato il suggerimento (e risposto in chat)

File: labs/lab5/lab5.md (hunk 4 above)
Comment (battibass) [reply labs/lab5/lab5.md]: operazioni

File: labs/lab5/lab5.md (hunk 1 above)
Comment (battibass) [reply labs/lab5/lab5.md]: interazioni

**Comment by battibass** [comment]
Grazie

**Pull request merged**